src/heap/src/Common.h

```diff
@@ -9,30 +9,49 @@
 extern "C" {
 #endif
 
-#pragma once
-#pragma pack(push, include, 1) // for byte alignment
-
 /**
  * 'Magic' - aka a sentinel which will be added as a guard band for each allocation in debug mode
  */
 #define ALLOCATION_HEADER_MAGIC "__HEADER_MAGIC_GUARD_BAND__"
 #define ALLOCATION_FOOTER_MAGIC "__FOOTER_MAGIC_GUARD_BAND__"
 
+// Ensure it's more than the actual header and footer size
+// IMPORTANT!!! This must be a multiple of 8 for alignment
+#define ALLOCATION_HEADER_MAGIC_SIZE        32
+#define ALLOCATION_FOOTER_MAGIC_SIZE        32
+
+/**
+ * The internal base structure representing the allocation header for
+ * all of the heap implementations.
+ *
+ * IMPORTANT!!! Ensure this structure is tightly packed without tight packing directives
+ */
 typedef struct
 {
+    // The aligned allocation size without the service structures - aka header and footer included
     UINT64 size;
+
+    // The type of the heap
     UINT32 type;
-    UINT32 handle;
+
+    // Union used for flags or VRAM handler
+    union {
+        UINT32 vramHandle;
+        UINT32 flags;
+    };
```

Comment:
why use union for two variables of same type? Also it's not clear how vramHandle cant co-exist with flags.

Reply:
The main idea here is the field name. For VRAM-based heap, it will use vramHandle. For AIV heap, it will be flags. Later, for file, it will be segment or something.